grpc.go

```diff
@@ -768,17 +768,22 @@ func (a *agentGRPC) WaitProcess(ctx context.Context, req *pb.WaitProcessRequest)
 		return &pb.WaitProcessResponse{}, err
 	}
 
-	defer func() {
+	defer proc.Do(func() {
 		proc.closePostExitFDs()
 		ctr.deleteProcess(proc.id)
-	}()
+	})
 
 	// Using helper function wait() to deal with the subreaper.
 	libContProcess := (*reaperLibcontainerProcess)(&(proc.process))
 	exitCode, err := a.sandbox.subreaper.wait(proc.exitCodeCh, libContProcess)
 	if err != nil {
 		return &pb.WaitProcessResponse{}, err
 	}
+	//refill the exitCodeCh with the exitcode which can be read out
+	//by another WaitProcess(). Since this channel isn't be closed,
+	//here the refill will always success and it will be free by GC
+	//once the process exits.
+	proc.exitCodeCh <- exitCode
```

Comment:
I think that we should still close the go channel from `deleteProcess()`.

Reply:
Hi @sboeuf , we cannot close the exitCodeCh in deleteProcess(),  if it's closed in deleteProcess(), then the other waitProcess() which are issued before the process exited
will not get the exit code, and this close will also cause the write panic in the next waitProcess().

Reply:
Ah yes, you're right.